tests/integrationv2/test_buffered_send.py

```diff
@@ -0,0 +1,161 @@
+import pytest
+
+from configuration import available_ports, PROTOCOLS, ALL_TEST_CIPHERS, Certificates
+from common import ProviderOptions, data_bytes
+from fixtures import managed_process # lgtm [py/unused-import]
+from providers import Provider, S2N, OpenSSL, GnuTLS
+from utils import invalid_test_parameters, get_parameter_name, to_bytes, is_subsequence
+
+SEND_DATA_SIZE = 2 ** 14
+
+K_BYTES = 1024 
+SEND_BUFFER_SIZE_MIN = 1031
+SEND_BUFFER_SIZE_MIN_RECOMMENDED = 2 * K_BYTES
+SEND_BUFFER_SIZE_MULTI_RECORD = 17 * K_BYTES
+SEND_BUFFER_SIZE_PREFER_THROUGHPUT = 35 * K_BYTES
+SEND_BUFFER_SIZE_HUGE = 512 * K_BYTES
+
+SEND_BUFFER_SIZES = [
+    SEND_BUFFER_SIZE_MIN,
+    SEND_BUFFER_SIZE_MIN_RECOMMENDED,
+    SEND_BUFFER_SIZE_MULTI_RECORD,
+    SEND_BUFFER_SIZE_PREFER_THROUGHPUT,
+    SEND_BUFFER_SIZE_HUGE
+]
+
+FRAGMENT_PREFERENCE = [
+    None,
+    "--prefer-low-latency",
+    "--prefer-throughput"
+]
+
+TEST_CERTS = [
+    Certificates.RSA_2048_SHA256,
+    Certificates.RSA_PSS_2048_SHA256,
+    Certificates.ECDSA_256
+]
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", ALL_TEST_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("other_provider", [GnuTLS, OpenSSL, S2N], ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [S2N], ids=get_parameter_name)
+@pytest.mark.parametrize("protocol", PROTOCOLS, ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("buffer_size", SEND_BUFFER_SIZES, ids=get_parameter_name)
+@pytest.mark.parametrize("fragment_preference", FRAGMENT_PREFERENCE, ids=get_parameter_name)
+def test_s2n_buffered_send_server(managed_process, cipher, other_provider, provider, protocol, certificate, buffer_size, fragment_preference):
+    # Communication Timeline
+    # Client [S2N|OpenSSL|GnuTLS]  | Server [S2N]
+    # Handshake                    | Handshake
+    #  Handshake finish indicated
+    #  by the client send marker
+    # Send Server Send Marker      | Receive the Server Send Marker
+    #                              | Send Data Bytes to Client
+    #                                 stresses the send buffer
+    #                                 center of test
+    #                              | Send Client Close Marker
+    # Close                        | Close
+
+    starting_client_send_marker = other_provider.get_send_marker()
+
+    starting_server_send_marker = "YTREWQ"
+    client_inital_app_data = to_bytes(starting_server_send_marker)
+    client_close_marker = server_sent_final = "QWERTY"
+    data_bytes_server = data_bytes(SEND_DATA_SIZE) + to_bytes(server_sent_final)
+
+    port = next(available_ports)
+
+    client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        port=port,
+        data_to_send=client_inital_app_data,
+        insecure=True,
+        protocol=protocol)
+
```

Comment:
Ah Ha! I misunderstood https://github.com/aws/s2n-tls/pull/3537#discussion_r991566738.

It isn't that s2n as the client (or as the server) can't set the cipher. S2N itself can't set the cipher.

Reply:
I could definitely have been clearer there. Yes, S2N can't set a specific cipher, since it works off of defined security policies. So we force a specific cipher via the peer's settings.

We generally set the cipher on both the client and server options, S2N just interprets it differently.

Reply:
Got it.